packages/twenty-front/src/modules/app/effect-components/PageChangeEffect.tsx

```diff
@@ -53,6 +62,29 @@ export const PageChangeEffect = () => {
     activityObjectNameSingular: CoreObjectNameSingular.Task,
   });
 
+  const [searchParams] = useSearchParams();
+  const { objectNameSingular, objectNamePlural } = useParams();
+
+  const objectMetadataItem = useRecoilValue(
+    objectMetadataItemFamilySelector(
+      objectNameSingular
+        ? { objectName: objectNameSingular, objectNameType: 'singular' }
+        : objectNamePlural
+          ? { objectName: objectNamePlural, objectNameType: 'plural' }
+          : { objectName: '', objectNameType: 'singular' },
+    ),
+  );
+
+  const setContextStoreCurrentViewId = useSetRecoilState(
```

Comment:
use this from viewId:QueryParamsViewIdEffect
and for objectMetadata, set it from RecordShow and RecordIndex

Reply:
+ reset from other pages